spec/view_objects/allocations_view_spec.rb

```diff
@@ -132,4 +132,187 @@
+
+        let(:training_providers) { [training_provider] }
+
+        let(:allocations) { [declined_allocation] }
+
+        it {
+          is_expected.to eq([{
+            training_provider_name: training_provider.provider_name,
+            training_provider_code: training_provider.provider_code,
+            status: AllocationsView::Status::NOT_REQUESTED,
+            status_colour: AllocationsView::Colour::RED,
+          }])
+        }
+      end
+
+      context "returns allocations where status is 'YET TO REQUEST'" do
+        let(:allocations) { [] }
+
+        let(:training_providers) { [training_provider] }
+
+        it {
+          is_expected.to eq([{
+            training_provider_name: training_provider.provider_name,
+            training_provider_code: training_provider.provider_code,
+            status: AllocationsView::Status::NO_REQUEST_SENT,
+            status_colour: AllocationsView::Colour::GREY,
+          }])
+        }
+      end
+
+      context "no allocations are returned if their status is 'REQUESTED" do
+        let(:repeat_allocation) do
+          build(:allocation, :repeat, accredited_body: accredited_body, provider: training_provider, number_of_places: 1)
+        end
+
+        let(:allocations) { [repeat_allocation] }
+
+        let(:training_providers) { [training_provider] }
+
+        it {
+          is_expected.to eq([])
+        }
+      end
+
+      context "no allocations are returned if their status is 'INITIAL'" do
+        let(:initial_allocation) do
+          build(:allocation, :initial, accredited_body: accredited_body, provider: training_provider, number_of_places: 1)
+        end
+
+        let(:allocations) { [initial_allocation] }
+
+        let(:training_providers) { [training_provider] }
+
+        it {
+          is_expected.to eq([])
+        }
+      end
+
+      context "check sort order is alphabetical" do
+        before do
+          training_provider.provider_name = "Training Provider A"
+          another_training_provider.provider_name = "Training Provider B"
+        end
+
+        let(:allocations) { [] }
+
+        it {
+          is_expected.to eq([{
+            training_provider_name: training_provider.provider_name,
+            training_provider_code: training_provider.provider_code,
+            status: AllocationsView::Status::NO_REQUEST_SENT,
+            status_colour: AllocationsView::Colour::GREY,
+          },
+                             {
+                               training_provider_name: another_training_provider.provider_name,
+                               training_provider_code: another_training_provider.provider_code,
+                               status: AllocationsView::Status::NO_REQUEST_SENT,
+                               status_colour: AllocationsView::Colour::GREY,
+                             }])
```

Comment:
ummm...


:space_invader:

Reply:
Yip 👽!!!

Don't not why it wasn't caught by the linter.